src/src/com/microsoft/aad/adal/IdToken.java

```diff
@@ -23,28 +23,145 @@
 
 package com.microsoft.aad.adal;
 
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.Iterator;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import android.util.Base64;
+
 class IdToken {
```

Comment:
@imakhalova Also, while it might be possible to refactor in the way that you I would consider that out of scope for this PR.  What we've done here is a pretty minor refactoring in order to make this code useable in more than the OAuth class.  It's really hard to draw the line on where to stop refactoring once you get started in this code, but I think we should take a deep breath, hold our nose, and let this one be.